### ISSUE TITLE
br restore log fails due to "scan region return empty result"

### ISSUE THREAD

**Issue by fubinzh** [pingcap/tidb#34750]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. Start a PiTR restore 
```
07:17:07 root@172 ~ → ./br_retry_pr restore point --pd 172.16.6.46:2379 --full-backup-storage 's3://nfs/fubin/pitr/pp2_full_0514?access-key=minioadmin&secret-access-key=minioadmin&endpoint=http%3a%2f%2fminio.pingcap.net%3a9000&force-path-style=true' --storage 's3://nfs/fubin/pitr/pp2_log_1?access-key=minioadmin&secret-access-key=minioadmin&endpoint=http%3a%2f%2fminio.pingcap.net%3a9000&force-path-style=true' --restored-ts "2022-05-15 18:15:00.000 +0800"
Detail BR log in /tmp/br.log.2022-05-17T19.17.12+0800
Full Restore <-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------> 100.00%
[2022/05/17 19:54:36.432 +08:00] [INFO] [collector.go:69] ["Full Restore success summary"] [total-ranges=8332] [ranges-succeed=8332] [ranges-failed=0] [split-region=4m22.082045564s] [restore-ranges=4690] [total-take=37m23.480771173s] [BackupTS=433268011009835010] [total-kv=1470368846] [total-kv-size=378.2GB] [average-speed=168.6MB/s] [restore-data-size(after-compressed)=210.7GB] [Size=210746803547]
Restore Meta Files <-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------> 100.00%
Restore KV Files <---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------> 100.00%
Error: failed to restore kv files: scan region return empty result, startKey: 7480000000000000FFCF5F698000000000FF0000020380000000FF1A3D490C00000000FB, endKey: 7480000000000000FFCF5F698000000000FF0000020380000000FF1B: [BR:PD:ErrPDBatchScanRegion]batch scan region
```

### 2. What did you expect to see? (Required)
Restore should be successful

### 3. What did you see instead (Required)
Restore failed due to "scan region return empty result"

### 4. What is your TiDB version? (Required)
08:35:03 root@172 ~ → ./br_retry_pr -V
Release Version: v6.1.0-alpha-425-gd3f8ed892
Git Commit Hash: d3f8ed892ab986c5d7e75bffa50de25e3f85b50e
Git Branch: restore_retry
Go Version: go1.18
UTC Build Time: 2022-05-16 10:19:37
Race Enabled: false

**Comment by joccau** [comment]
This issue is fixed by the PR(https://github.com/pingcap/tidb/pull/34776)

**Comment by windtalker** [comment]
Remove `affects-6.1` label since the fix is already in 6.1
